internal/provider/providers/porkbun/api.go

```diff
@@ -32,54 +42,48 @@ func (p *Provider) getRecordIDs(ctx context.Context, client *http.Client, record
 	encoder := json.NewEncoder(buffer)
 	err = encoder.Encode(postRecordsParams)
 	if err != nil {
-		return nil, fmt.Errorf("json encoding request data: %w", err)
+		return nil, fmt.Errorf("json encoding %s request data: %w", recordType, err)
 	}
 
 	encodedRequestURL := u.String()
-	if p.owner != "@" {
+	if owner != "@" {
 		// add owner after string-ing the URL to avoid encoding the '*' character,
 		// since Porkbun requires we send the unencoded '*' character.
-		encodedRequestURL += p.owner
+		encodedRequestURL += owner
 	}
 
 	request, err := http.NewRequestWithContext(ctx, http.MethodPost, encodedRequestURL, buffer)
 	if err != nil {
-		return nil, fmt.Errorf("creating http request: %w", err)
+		return nil, fmt.Errorf("creating %s http request: %w", recordType, err)
 	}
 	setHeaders(request)
 
 	response, err := client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("doing http request: %w", err)
+		return nil, fmt.Errorf("doing %s http request: %w", recordType, err)
 	}
 	defer response.Body.Close()
 
 	if response.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("%w: %d: %s", errors.ErrHTTPStatusNotValid,
+		return nil, fmt.Errorf("%s - %w: %d: %s", recordType, errors.ErrHTTPStatusNotValid,
```

Comment:
Wasn't entirely sure the best way to format these error messages to include the recordType. I welcome any input